### PR TITLE
Rationalising button variants and adding Notes.

### LINKED ISSUE #17: Add additional button variants
The button component needs the following variants adding:

* btn--lg
* btn--loader

It requires documentation detailing the purpose and rationale for each variant.

### PR DESCRIPTION
**Pull request by armstrongb**
### What is the context of this PR?

- Rationalised available button variants from a `.hbs` file per variant to all variants contained in the config.js file and using `button.hbs` as a template.
- Added `readme.md` draft notes about 'Button'.

### How to to test

Pull branch, run `yarn start` and use `localhost:3000` to browse to the button component.

### Issues

Resolves #17

### CODE REVIEW

File: components/button/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+### Rationale
+Buttons are used to direct the user to perform a specific interaction. The copy used on the button should be clear, concise and direct.
+
+## Variants
+
```
Comment (shahi645) [inline components/button/README.md]: @armstrongb Could add all the variants of button style here. Although they are listed below

File: yarn.lock [hunk 2]
```diff
@@ -1,7 +1,5 @@
 # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (shahi645) [inline yarn.lock]: @armstrongb I dont think we are supposed to commit the `yarn.lock` are we?

Comment (benjefferies) [reply yarn.lock]: https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control

Comment (shahi645) [reply yarn.lock]: I stand corrected. Thanks @benjefferies

File: components/button/button.config.js [hunk 3]
```diff
@@ -1,6 +1,42 @@
...
+    "label": "Secondary - bordered",
+    "context": {
+      "btn-classes": "btn--secondary btn--border",
+      "label": "Add a person"
+    },
+  }, {
+    "name": "link",
+    "label": "Link",
+    "context": {
+      "btn-classes": "btn--link",
```
Comment (jonnyshaw89) [inline components/button/button.config.js]: I thought Buttons styles as links was something we wanted to avoid?

Comment (shahi645) [reply components/button/button.config.js]: Yeh but it is a current pattern. This PR is about implementing buttons in a better way to the pattern library

**Comment by hamishtaplin** [comment]
screenshot 2017-12-04 09 18 54

Link looks a bit weird so close to the next item, can that be fixed without doing anything weird in the template?

**Comment by shahi645** [comment]
maybe if we put the link after the last button? There is still a gap but looks a little better
screen shot 2017-12-04 at 09 29 46

**Comment by armstrongb** [comment]
@shahi645 this has been done now.

**Comment by ajmaddaford** [comment]
@armstrongb can this be squashed and merged now?

**Pull request merged**